aws_lambda_powertools/utilities/idempotency/__init__.py

```diff
@@ -2,6 +2,7 @@
 Utility for adding idempotency to lambda functions
 """
 
+from aws_lambda_powertools.utilities.idempotency.config import IdempotentHookData, IdempotentHookFunction
```

Comment:
Would it be better to move these classes to `hook.py` or something like that?

Reply:
Maybe let's jump on a call - placed where they are to avoid circular references between base.py, config.py and hook.py - happy for guidance  to improve - that's where they started.

Reply:
Sure thing! Let's schedule a call to review and potentially merge this PR before the next release. I'll ping you on Discard and we can find a slot to join in a meeting.

Just out of curiosity, why would we have issues with circular references here? We're importing from different files, so it shouldn't be a problem. I might be missing something, but I can't see any potential problems with circular references here.

I've created a patch with this modification, and all tests are passing, mypy is also happy, and I couldn't spot any issues with circular references. 
Could you please try applying this patch in your local environment?

[hook_patch.txt](https://github.com/aws-powertools/powertools-lambda-python/files/14844033/hook_patch.txt)